src/Common/tests/System/Runtime/Serialization/Utils.cs

```diff
@@ -73,6 +73,7 @@ private static CompareResult CompareString(string expected, string actual)
             return new CompareResult { ErrorMessage = "actual was null, but was expecting:\n" + expected };
         }
 
+        actual = actual.Replace('\u00A0', ' '); // normalize non breaking space to space; varies by OS
```

Comment:
I'm not understanding this change.

Are we doing this because we accidentally have some non-breaking spaces in our test files?  If so, shouldn't we be fixing those instead of adding this here?

Or are we doing this because we explicitly have non-breaking spaces as test inputs?  If so, isn't this actually hampering testing we intended to be doing?  Even if we have to do this for certain OSes, why force it for all?

Reply:
> Are we doing this because we accidentally have some non-breaking spaces in our test files?
> Or are we doing this because we explicitly have non-breaking spaces as test inputs? 

We have no nbsp as input, the string `actual` is populated using `DataContractJsonSerializerSettings` which somehow adds nbsp into date formated string. 

>  Even if we have to do this for certain OSes, why force it for all? 

The date string populated having no any nbsp for other OSes. Somehow it started adding that nbsp for Catalina. I wouldn't name it its forcing it for all OS, it is normalizing different output of space into same ...

Reply:
> Somehow it started adding that nbsp for Catalina

Can we figure out why it's happening and whether it's a problem, rather than hiding it in the tests?  If there's actually a problem here, presumably customers could hit that, too?

Or have we already investigated?

Reply:
I didn't really dig into it that deep, I agree its better to dig into it more and will lookup as much as I can. Also I would like to hear area owner's opinion CC @StephenMolloy @HongGit

Reply:
@stephentoub the issue is DateTime.ToString(string format, IFormatProvider provider) is producing non breaking space between `a. m.`. Following code produce different result for windows/Mojave vs Catalina

```csharp
DateTime value = new DateTime();
string val = value.ToString("hh:mm:ss.ff tt", new DateTimeFormat("hh:mm:ss.ff tt", CultureInfo.CreateSpecificCulture("es-AR")).FormatProvider);
Console.WriteLine(val + ", "+ (int)val.ToCharArray()[11] + ", "+(int)val.ToCharArray()[14]);
```
for windows or Mojave the result: `12:00:00.00 a. m., 32, 32`
but for Catalina it is `12:00:00.00 a. m., 32, 160`

I assume it could cause issue to customers so am gonna report a bug about that. But for this fix i think we could go with it until the bug is resolved. What you think?

Reply:
Turns out it is not an issue, for Osx 10.15 ICU updated to have CultureInfo..DateTimeFormat.AMDesignator string to have non breaking space which was normal space for previous version. So we shouldn't hard code comparing string `expected` with `a. m.`. It should loaded from the system dynamically. Thank you @tarekgh for explanation, fix coming soon

Reply:
Updated, please re review. Thank you for pointing it @stephentoub

Reply:
Thank you for investigating and fixing it in a more targeted manner.